internal/pkg/config/import_services.go

```diff
@@ -1,5 +1,16 @@
 package config
 
-type ImportedServices struct {
+import (
+	istionetv1alpha3 "istio.io/api/networking/v1alpha3"
+)
+
+type ImportedServiceSet struct {
 	Rules []Rules `yaml:"rules"`
 }
+
+type ImportedService struct {
+	AppName         string
+	ServiceHostname string
+	ServicePorts    []*istionetv1alpha3.ServicePort
```

Comment:
I wonder how we can obtain Istio ports if we watch k8s services...? It seems that we will have to require specifying `appProtocol` in exported services. Only then we are able to translate it to proper Istio protocol.

Reply:
Indeed I was thinking we would likely need to export the port protocol from the remote service. Either that or we could perhaps use some similar logic that Istio uses to determine the port protocol from the port name: https://github.com/istio/istio/blob/master/pkg/config/kube/conversion.go#L47-L87

But probably better still to explicitly export the port.